src/sumo_logic.js

```diff
@@ -3,71 +3,73 @@ import $ from "jquery";
 const SYNC_INTERVAL = 5000;
 
 class SumoLogic {
-  constructor(settings={}) {
+  constructor(settings= {}, context = {}) {
     this.settings = settings;
     this.messages = [];
-  }
-  
-  static log(msg) {
-    this.logger = this.logger || new SumoLogic(this.settings);
-    this.logger.log(msg);
+    this.context = context;
+    this.syncInterval = SYNC_INTERVAL;
+    if (settings.captureConsole === undefined || settings.captureConsole) {
+      this.captureConsole();
+    }
+    if (settings.captureError === undefined  || settings.captureError) {
+      this.captureError();
+    }
   }
 
-  static dump(cb){
-    this.logger.dump(cb);
+ log(msg) {
+    this._addMessage(msg);
   }
 
-  static set settings(new_settings) {
-    this._settings = new_settings;
+  error(msg) {
+    this._addMessage(msg, 'error');
   }
 
-  static get settings() {
-    return this._settings;
+  warn(msg) {
+    this._addMessage(msg, 'warning');
   }
 
-  static set context(context){
-    this._context = context;
+  info(msg) {
+    this._addMessage(msg);
   }
 
-  static get context(){
-    return this._context;
-  }
+  dump(success_cb){
+    if(this.messages.length == 0) return;
 
-  get syncInterval() {
-    return this.settings.syncInterval || SYNC_INTERVAL;
+    this.sendMessages().done((response)=> this.onMessagesSent(response, success_cb));
   }
 
-  log(msg) {
-    this.validateSettings();
-    this.addMessage(msg);
-    if(!this.intervalId){
-      this.startDumping();
+  _addMessage(message, level = 'info') {
```

Comment:
My objective is to make a clear distinction between the public api and private methods.